### PR TITLE
nrfutil: 5.2 -> 6.1, pc-ble-driver-py: 0.11.4 -> 0.14.2, py-ble-driver: init at 4.1.1

### PR DESCRIPTION
**Pull request by jonas-schievink**
###### Motivation for this change

Fix the build of nrfutil by updating it and its dependencies.

cc @gebner

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pc-ble-driver-py/default.nix [hunk 1]
```diff
@@ -1,45 +1,24 @@
-{ stdenv, buildPythonPackage, fetchpatch, fetchFromGitHub,
-  python, cmake, git, swig, boost, udev,
-  setuptools, enum34, wrapt, future }:
+{ stdenv, fetchFromGitHub, cmake, git, swig, boost, udev, pc-ble-driver, python37Packages }:
 
-buildPythonPackage rec {
+with python37Packages; buildPythonPackage rec {
```
Comment (gebner) [inline pkgs/development/python-modules/pc-ble-driver-py/default.nix:3]:
```suggestion
buildPythonPackage rec {
```
Python packages can be built for different python versions.  For example `nix-build -A python38Packages.pc-ble-driver-py`.  This file should not import `python37Packages`.

**Review by gebner — CHANGES_REQUESTED** [review]
Otherwise LGTM.

File: pkgs/development/python-modules/pc-ble-driver-py/default.nix (hunk 1 above)
Comment (jonas-schievink) [reply pkgs/development/python-modules/pc-ble-driver-py/default.nix]: Fixed. Same for nrfutil? It doesn't build on all Python 3 versions, and not at all on Python 2.

Comment (gebner) [reply pkgs/development/python-modules/pc-ble-driver-py/default.nix]: For applications it's fine to hardcode a python version.  It's only the libraries where you might want to use them with a different python version.

Please squash the "Don't depend on a fixed Python version" commit into the "nrfutil: ..." commit.

Comment (jonas-schievink) [reply pkgs/development/python-modules/pc-ble-driver-py/default.nix]: Ok, done! (I'm assuming you meant the pc-ble-driver-py commit)

Comment (gebner) [reply pkgs/development/python-modules/pc-ble-driver-py/default.nix]: Oh yes, of course.

**Comment by gebner** [comment]
Thank you!

**Pull request merged**